debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -183,6 +190,33 @@ public BinlogReader(String name, MySqlTaskContext context, HaltingPredicate acce
         client.setThreadFactory(Threads.threadFactory(MySqlConnector.class, context.getConnectorConfig().getLogicalName(), "binlog-client", false));
         client.setServerId(serverId);
         client.setSSLMode(sslModeFor(connectionContext.sslMode()));
+        if (connectionContext.sslModeEnabled() &&
+            connectionContext.jdbc().config().getString(MySqlJdbcContext.JDBC_PROPERTY_ENABLE_TLS_PROTOCOLS) != null) {
+            client.setSslSocketFactory(new DefaultSSLSocketFactory(
+                connectionContext.jdbc().config().getString(MySqlJdbcContext.JDBC_PROPERTY_ENABLE_TLS_PROTOCOLS)) {
+
+                @Override
+                protected void initSSLContext(SSLContext sc) throws GeneralSecurityException {
+                    sc.init(null, new TrustManager[]{
+                        new X509TrustManager() {
+
+                            @Override
+                            public void checkClientTrusted(X509Certificate[] x509Certificates, String s)
+                                throws CertificateException { }
```

Comment:
I think this goes back to your last comment in the PR description, but doesn't this conflict with "strict" SSL checking?

Reply:
@jordanbragg Please take a look at https://github.com/shyiko/mysql-binlog-connector-java/blob/master/src/main/java/com/github/shyiko/mysql/binlog/BinaryLogClient.java#L704
You should have two factories and use theme depending on value of `sslMode` property.